pkg/kubelet/dockertools/manager.go

```diff
@@ -1893,7 +1894,10 @@ func (dm *DockerManager) SyncPod(pod *api.Pod, runningPod kubecontainer.Pod, pod
 
 		// Find the pod IP after starting the infra container in order to expose
 		// it safely via the downward API without a race and be able to use podIP in kubelet-managed /etc/hosts file.
-		pod.Status.PodIP = dm.determineContainerIP(pod.Name, pod.Namespace, podInfraContainer)
+		pod.Status.PodIP, err = dm.determineContainerIP(pod.Name, pod.Namespace, podInfraContainer)
+		if err != nil {
+			return err
```

Comment:
Inspect failures could be transient. Unless we keep track and confirm that it consistently fails, it's probably not the best to kill it right away?